File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isAlphaNum(char c){
        if ((int('A') <= int(c) && int(c) <= int('Z')) or (int('a') <= int(c) && int(c) <= int('z')) or (int('0') <= int(c) && int(c) <= int('9'))) return true;
        return false;
    }

    bool isPalindrome(string s) {
        int i = 0;
        int j = s.length() - 1;
        while (i < j){
            while ((i < j) && !isalnum(s[i])) i += 1;
            while ((i < j) && !isalnum(s[j])) j -= 1;
            if(tolower(s[i]) != tolower(s[j])) return false;
            i += 1;
            j -= 1;
        }
        return true;
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isAlphaNum(self, str):
        if (ord('A') <= ord(str) <= ord('Z')) or (ord('a') <= ord(str) <= ord('z')) or ord('0') <= ord(str) <= ord('9'):
            return True
        return False

    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        while i<j:
            while i<j and not s[i].isalnum():
                i += 1
            while i<j and not s[j].isalnum():
                j -= 1
            if s[i].lower() != s[j].lower():
                return False
            i += 1
            j -= 1
        return True